src/iOS.Autofill/LoginListViewController.cs

```diff
@@ -306,62 +470,53 @@ protected override void Dispose(bool disposing)
             base.Dispose(disposing);
         }
 
-        public class TableSource : BaseLoginListTableSource<LoginListViewController>
+        private async Task LoadSourceAsync()
         {
-            public TableSource(LoginListViewController controller)
-                : base(controller)
-            {
-            }
-
-            protected override string LoginAddSegue => SegueConstants.ADD_LOGIN;
+            _isLoading = true;
 
-            public override async Task LoadAsync(bool urlFilter = true, string searchFilter = null)
+            try
             {
-                try
+                await MainThread.InvokeOnMainThreadAsync(() =>
                 {
-                    await base.LoadAsync(urlFilter, searchFilter);
+                    TableView.Hidden = true;
+                    _searchBar.Hidden = true;
+                    _loadingView.Hidden = false;
+                });
 
-                    if (Context.IsCreatingPasskey && !Items.Any())
-                    {
-                        Controller?.OnEmptyList();
-                    }
-                }
-                catch (Exception ex)
+                await Source.LoadAsync(string.IsNullOrEmpty(_searchBar?.Text), _searchBar?.Text);
+
+                await MainThread.InvokeOnMainThreadAsync(() =>
                 {
-                    LoggerHelper.LogEvenIfCantBeResolved(ex);
-                }
+                    _loadingView.Hidden = true;
+                    TableView.Hidden = Context.IsCreatingPasskey && Source.IsEmpty;
+                    _searchBar.Hidden = string.IsNullOrEmpty(_searchBar?.Text);
+                });
             }
-
-            public override UIView GetViewForHeader(UITableView tableView, nint section)
+            finally
             {
-                try
-                {
-                    if (Context.IsCreatingPasskey
-                        &&
-                        tableView.DequeueReusableHeaderFooterView(LoginListViewController.HEADER_SECTION_IDENTIFIER) is HeaderItemView headerItemView)
-                    {
-                        headerItemView.SetHeaderText(AppResources.ChooseALoginToSaveThisPasskeyTo);
-                        return headerItemView;
-                    }
-
-                    return new UIView(CGRect.Empty);
-                }
-                catch (Exception ex)
-                {
-                    LoggerHelper.LogEvenIfCantBeResolved(ex);
-                    return new UIView();
-                }
+                _isLoading = false;
```

Comment:
❓ If there is an exception we are currently only defaulting the `_isLoading`, should we also change the `Hidden` property of `TableView`, `_searchBar` and `_loadingView`?

Reply:
Hmmm, if there's an exception there then the view would be in a corrupted state, so I wouldn't "un-hide" the other views. I'll add an alert to the user with a general error message; thanks for bringing this up.